### PR TITLE
tilt: pass service defs to daemon

### PR DESCRIPTION
**Pull request by landism**

### CODE REVIEW

File: internal/tiltfile/tiltfile.go [hunk 1]
```diff
@@ -25,3 +26,34 @@ func Load(filename string) (*Tiltfile, error) {
 
 	return &Tiltfile{globals, filename, thread}, nil
 }
+
+func (tiltfile Tiltfile) GetServiceConfig(serviceName string) (*string, error) {
```
Comment (jazzdan) [inline internal/tiltfile/tiltfile.go]: String pointer! Exciting :)

File: internal/tiltfile/tiltfile_test.go [hunk 2]
```diff
@@ -34,8 +35,72 @@ def blorgly_frontend():
...
 	defer os.Remove(file)
 	tiltConfig, err := Load(file)
-	for _, s := range([]string{"blorgly", "blorgly_backend", "blorgly_frontend"}) {
+	for _, s := range []string{"blorgly", "blorgly_backend", "blorgly_frontend"} {
 		assert.Contains(t, tiltConfig.globals, s)
 	}
 	assert.Nil(t, err)
 }
+
+func TestGetServiceConfig(t *testing.T) {
```
Comment (jazzdan) [inline internal/tiltfile/tiltfile_test.go]: yasss tests!

File: internal/tiltfile/tiltfile.go [hunk 3]
```diff
@@ -25,3 +26,34 @@ func Load(filename string) (*Tiltfile, error) {
...
+	f, ok := tiltfile.globals[serviceName]
+
+	if !ok {
+		return nil, errors.New(fmt.Sprintf("%v does not define a global named '%v'", tiltfile.filename, serviceName))
```
Comment (nicks) [inline internal/tiltfile/tiltfile.go]: fwiw, there's a shortcut function `fmt.Errorf`

File: internal/tiltfile/tiltfile.go (hunk 1 above)
Comment (nicks) [reply internal/tiltfile/tiltfile.go]: ya, this seems non-idiomatic. most functions like this return (string, error) and return the empty string on error

**Pull request merged**